spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/convert/SqlGenerator.java

```diff
@@ -783,12 +783,51 @@ private List<OrderByField> extractOrderByFields(Sort sort) {
 	}
 
 	private OrderByField orderToOrderByField(Sort.Order order) {
-
-		SqlIdentifier columnName = this.entity.getRequiredPersistentProperty(order.getProperty()).getColumnName();
+		SqlIdentifier columnName = getColumnNameToSortBy(order);
 		Column column = Column.create(columnName, this.getTable());
 		return OrderByField.from(column, order.getDirection()).withNullHandling(order.getNullHandling());
 	}
 
+	private SqlIdentifier getColumnNameToSortBy(Sort.Order order) {
```

Comment:
This should be done using the `PropertyPath` abstraction. See https://github.com/spring-projects/spring-data-commons/blob/main/src/main/java/org/springframework/data/mapping/context/MappingContext.java#L174

Reply:
I have utilized `PersistentPropertyPath` for that, I guess it will be even more accurate here